rpcs3/Emu/Io/usb_device.h

```diff
@@ -153,15 +154,17 @@ struct UsbDescriptorNode
 		}
 		return nodesize;
 	}
-	void write_data(u8*& ptr)
+
+	void write_data(u8*& ptr, u32& max_size) const
```

Comment:
Also, if you are already rewriting this, these references is not a good practice. I suggest return the size of the written data as the return value of the function, leaving the choice to modify the pointer and size up to the caller by the amout if written data.

Reply:
yes. either that or a wrapper that creates the pointer copy internally.

Reply:
No problem to me. I further optimized it.
I validated it with a few games, and everything worked fine as usual.